internal/tools/orchestrator/services/runtime/runtime.go

```diff
@@ -2081,11 +2081,22 @@ func (r *Runtime) fullGCForSingleRuntime(runtimeID uint64, keep int) {
 		return
 	}
 	oldestID := top[len(top)-1].ID
+	var hasSuccess bool
+	for _, deployment := range top {
```

Comment:
Is it possible that`top` latest history may all failed, causing latest success to be ignored.

Reply:
if the latest success  isn't in the `keep` list, it will be skipped when deployments are marked `out date`.